### ISSUE TITLE
Can dotnet store command exclude dependencies from Microsoft.NETCore.App?

### ISSUE THREAD

**Issue by normj** [dotnet/cli#10784]
## Steps to reproduce

1) Create an xml file listing a package that depends on packages included in Microsoft.NETCore.App. For example.

```xml
<Project Sdk="Microsoft.NET.Sdk">
  <ItemGroup>
    <PackageReference Include="Swashbuckle.AspNetCore.Swagger" Version="4.0.1" />
  </ItemGroup>
</Project>
```

2) Run the `dotnet store` command
```
dotnet store -m test-manifest.xml -o ./store --skip-optimization -r rhel.7.2-x64 --framework netcoreapp2.1
```
## Expected  behavior

In the runtime package store I expect to see just packages that are not included as part of the .NET Core install including the ones delivered with Microsoft.NETCore.App.

Hoping for an artifact.xml file like this:
```xml
<StoreArtifacts>
  <Package Id="Swashbuckle.AspNetCore.Swagger" Version="4.0.1" />
</StoreArtifacts>
```

## Actual behavior

The store includes all of the ASP.NET Core dependencies as bloating my store because those packages are already on the target environment as part of the .NET Core install of Microsoft.NETCore.App.

What I'm looking for is if there is a way to communicate through either the `dotnet store` command or the manifest that this store is being created for an environment that has Microsoft.NETCore.App installed so ignore dependencies that would come from it.

artifacts.xml file:
```xml
<StoreArtifacts>
  <Package Id="Microsoft.AspNetCore.Authentication.Abstractions" Version="2.0.0" />
  <Package Id="Microsoft.AspNetCore.Authentication.Core" Version="2.0.0" />
  <Package Id="Microsoft.AspNetCore.Authorization" Version="2.0.0" />
  <Package Id="Microsoft.AspNetCore.Authorization.Policy" Version="2.0.0" />
  <Package Id="Microsoft.AspNetCore.Hosting.Abstractions" Version="2.0.0" />
  <Package Id="Microsoft.AspNetCore.Hosting.Server.Abstractions" Version="2.0.0" />
  <Package Id="Microsoft.AspNetCore.Http" Version="2.0.0" />
  <Package Id="Microsoft.AspNetCore.Http.Abstractions" Version="2.0.0" />
  <Package Id="Microsoft.AspNetCore.Http.Extensions" Version="2.0.0" />
  <Package Id="Microsoft.AspNetCore.Http.Features" Version="2.0.0" />
  <Package Id="Microsoft.AspNetCore.JsonPatch" Version="2.0.0" />
  <Package Id="Microsoft.AspNetCore.Mvc.Abstractions" Version="2.0.0" />
  <Package Id="Microsoft.AspNetCore.Mvc.Core" Version="2.0.0" />
  <Package Id="Microsoft.AspNetCore.Mvc.Formatters.Json" Version="2.0.0" />
  <Package Id="Microsoft.AspNetCore.ResponseCaching.Abstractions" Version="2.0.0" />
  <Package Id="Microsoft.AspNetCore.Routing" Version="2.0.0" />
  <Package Id="Microsoft.AspNetCore.Routing.Abstractions" Version="2.0.0" />
  <Package Id="Microsoft.AspNetCore.WebUtilities" Version="2.0.0" />
  <Package Id="Microsoft.DotNet.PlatformAbstractions" Version="2.0.0" />
  <Package Id="Microsoft.Extensions.Configuration.Abstractions" Version="2.0.0" />
  <Package Id="Microsoft.Extensions.DependencyInjection.Abstractions" Version="2.0.0" />
  <Package Id="Microsoft.Extensions.DependencyModel" Version="2.0.0" />
  <Package Id="Microsoft.Extensions.FileProviders.Abstractions" Version="2.0.0" />
  <Package Id="Microsoft.Extensions.Hosting.Abstractions" Version="2.0.0" />
  <Package Id="Microsoft.Extensions.Logging.Abstractions" Version="2.0.0" />
  <Package Id="Microsoft.Extensions.ObjectPool" Version="2.0.0" />
  <Package Id="Microsoft.Extensions.Options" Version="2.0.0" />
  <Package Id="Microsoft.Extensions.Primitives" Version="2.0.0" />
  <Package Id="Microsoft.Net.Http.Headers" Version="2.0.0" />
  <Package Id="Newtonsoft.Json" Version="10.0.1" />
  <Package Id="Swashbuckle.AspNetCore.Swagger" Version="4.0.1" />
  <Package Id="System.Runtime.CompilerServices.Unsafe" Version="4.4.0" />
  <Package Id="System.Text.Encodings.Web" Version="4.4.0" />
</StoreArtifacts>
```

## Environment data
~/temp/LambdaLayer/AspNetCoreTest/src/AspNetCoreTest$ dotnet --info
.NET Core SDK (reflecting any global.json):
 Version:   2.2.100
 Commit:    51868761f2

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  10.14
 OS Platform: Darwin
 RID:         osx.10.14-x64
 Base Path:   /usr/local/share/dotnet/sdk/2.2.100/

Host (useful for support):
  Version: 2.2.0
  Commit:  1249f08fed

.NET Core SDKs installed:
  1.1.11 [/usr/local/share/dotnet/sdk]
  2.1.302 [/usr/local/share/dotnet/sdk]
  2.1.403 [/usr/local/share/dotnet/sdk]
  2.1.503 [/usr/local/share/dotnet/sdk]
  2.2.100 [/usr/local/share/dotnet/sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.2 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.5 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.7 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.2.0 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.2 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.5 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.7 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.2.0 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 1.0.13 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 1.1.10 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.2 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.5 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.7 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.2.0 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]

**Comment by nguerrera** [comment]
Yes, this is possible. 

First thing to note is that Microsoft.NETCore.App assemblies are actually excluded. The dependencies you're seeing are actually provided by Microsoft.**AspNet**Core.App shared framework, and not the base Microsoft.NETCore.App. 

If you pull in Microsoft.AspNetCore.App, then the system will see the additional assemblies it provides and will not store them. You can do that just as you would in a 2.x web csproj file by specifying Microsoft.AspNetCore.App as a versionless package reference. 

That should be all that is needed, but it turns out https://github.com/dotnet/cli/issues/10602 also causes extra files to be stored as it causes extra files to be published. It will be fixed in the next 2.2.1xx SDK patch, but it can be worked around rather easily in this case in the meantime.

The following test manifest should do what you want with SDK 2.2.100+:
```
<Project Sdk="Microsoft.NET.Sdk">
  <ItemGroup>
    <PackageReference Include="Swashbuckle.AspNetCore.Swagger" Version="4.0.1" />

    <!-- Pull in Microsoft.AspNetCore.App so that the system can see it as the platform and
         excludes dependencies of above from being restored if they are provided by it. -->
    <PackageReference Include="Microsoft.AspNetCore.App" />

    <!-- This shouldn't be necessary, but it works around https://github.com/dotnet/cli/issues/10602 -->
    <!-- The issue will be fixed in the next 2.2.1xx SDK patch and this line can be removed then.-->
    <PackageReference Update="Microsoft.NETCore.App" Publish="false" />
  </ItemGroup>
</Project>
```

**Comment by normj** [comment]
Thanks for the insight. It worked perfectly.
